.github/workflows/check-learning-path-links.yml

```diff
@@ -0,0 +1,115 @@
+          cd ../
+
+      - name: Checkout head
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: true
+          ref: main
+
+      - name: Checkout previous update
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          ref:  ${{ env.prev_sha }}
+          path: prev
+
+      - name: Get changed files
+        run: |
+          changed_source_files=$(git diff-tree --no-commit-id --name-only -r "$prev_sha" "$GITHUB_SHA" | { grep "**.cs$" || test $? = 1; })
+          echo "Files to validate: '${changed_source_files}'"
+          echo "updated_files=$(echo ${changed_source_files})" >> $GITHUB_ENV
+
+      - name: Check Learning Path Links
+        id: check-links
+        uses: ./.github/actions/learning-path-staleness-check
+        with:
+          repoURLToSearch: 'https://github.com/dotnet/dotnet-monitor'
+          learningPathsDirectory: 'documentation/learningPath'
+          changedFilePaths: ${{ env.updated_files }}
+          sourceDirectoryName: 'src'
+          oldHash: ${{ env.prev_sha }}
+          newHash: ${{ env.new_sha }}
+          learningPathHashFile: '.github/learning-path-sha.txt'
+
+      - name: Create PR Body
+        run: |
+          manuallyReview=$MANUALLY_REVIEW
+          manuallyReview=${manuallyReview//,/<br />}
+          manuallyReviewSection="<h2>Manually Review:</h2>${manuallyReview}<br />"
+          if [ -z "$manuallyReview" ]; then
+            manuallyReviewSection=""
+          fi
+          outOfSync=$OUT_OF_SYNC
+          outOfSync=${outOfSync//,/<br />}
+          outOfSyncSection="<h2>Links With Out Of Sync Commit Hashes:</h2>${outOfSync}<br />"
+          if [ -z "$outOfSync" ]; then
+            outOfSyncSection=""
+          fi
+          suggestions=$SUGGESTIONS
+          suggestions=${suggestions//,/<br />}
+          suggestionsSection="<h2>Auto-Applied Suggestions:</h2>${suggestions}<br />"
+          if [ -z "$suggestions" ]; then
+            suggestionsSection=""
+          fi
+          modifiedFiles=$MODIFIED_FILES
+          modifiedFiles=${modifiedFiles//,/<br />}
+          modifiedFilesSection="<h2>Modified Files:</h2>${modifiedFiles}<br />"
+          if [ -z "$modifiedFiles" ]; then
+            modifiedFilesSection=""
+          fi
+          body="${modifiedFilesSection}${manuallyReviewSection}${outOfSyncSection}${suggestionsSection}"
+          echo "body=$body" >> $GITHUB_ENV
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          MODIFIED_FILES: ${{ steps.check-links.outputs.modifiedFiles }}
+          MANUALLY_REVIEW: ${{ steps.check-links.outputs.manuallyReview }}
+          SUGGESTIONS: ${{ steps.check-links.outputs.suggestions }}
+          OUT_OF_SYNC: ${{ steps.check-links.outputs.outOfSync }}
+
+      - name: Open PR
+        uses: ./.github/actions/open-pr
+        with:
+          files_to_commit: ${{ steps.check-links.outputs.modifiedFilesToCommit }}
```

Comment:
We shouldn't actually need this output from the GitHub action, we can just tell this `open-pr` action to commit any files that have changed after `check-link` was ran.

e.g.


```yaml
      - name: Open PR
        uses: ./.github/actions/open-pr
        with:
          files_to_commit: "*"
```

Reply:
I did try this for a run, but it appears that because I have two versions of the repo checked out (the main and the "prev"), it wasn't happy with the `*` (here's the run for reference https://github.com/kkeirstead/dotnet-monitor/actions/runs/7743591239/job/21115353520). If we want to avoid the current solution I can look into this more, but figured I'd wait until you weighed in.

Reply:
From what I can tell this is because the "prev" repo is being check out inside of "main" as opposed to next to it

How it looks now:
```
-> /home/runner/work/dotnet-monitor/
--> /dotnet-monitor (this is "main")
-----> /prev
```

What it should be:
```
-> /home/runner/work/dotnet-monitor/
--> /dotnet-monitor
--> /prev
```

If you would like, I can take a look at the options for actions/checkout to see how we should configure it

Reply:
So, there is a way to check out side-by-side (that was what I originally was doing before I switched to this approach), but then I was having issues with the open-pr action not working. I spent most of this morning trying to get this to work, but I'm still hitting snags where I get close but it's just not quite right. I don't doubt there's a way to do this - I just don't think it's worth me spending more time on it right now for the same end result.

IMO, unless there's still more security concerns, I'm really trying to wrap up this PR. There's been so much churn from the original prototype that this has taken way more time than I wanted to allocate to it. We can iterate on it as-needed for any edge-cases I missed or other optimizations to make it cleaner or work better.

Reply:
> We can iterate on it as-needed for any edge-cases I missed or other optimizations to make it cleaner or work better.

That sounds good to me. As a short term solution you should just be able to use `--all -- :!prev` as the `files_to_commit`, which will add everything except files from the `prev` folder